WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -360,12 +378,41 @@ public void onPluginDirectoryFetched(PluginStore.OnPluginDirectoryFetched event)
             }
         }
 
+        PluginListType listType = event.type == PluginDirectoryType.POPULAR ?
+                PluginListType.POPULAR : PluginListType.NEW;
         PluginListFragment fragment = getListFragment();
-        if (fragment != null) {
+        if (fragment != null && fragment.getListType() == listType) {
             fragment.onLoadedMore(event.canLoadMore);
```

Comment:
I know this is from a previous PR, but I didn't get a chance to review that, so I want to make sure to mention it before I forget it. I feel like this whole section should be under the `else`, because I don't think we should be handling both the error and the success cases together.

Concretely, I think the plugin directory fetch can fail temporarily in which case since we are passing the `event.canLoadMore` it'll always stop asking for more data. Which might be something we want in any case, but it feels weird that it's a consequence instead of a direct decision.

I've no idea if this explanation makes sense at all. I don't think it's too big deal of a deal, so feel free to ignore it if it doesn't make sense.